packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -0,0 +1,204 @@
+    await this.messagingSystem.call('NetworkController:setActiveNetwork', id);
+  }
+
+  /**
+   * Sets the active non-EVM network.
+   *
+   * @param id - The chain ID of the non-EVM network to set active.
+   */
+  #setActiveNonEvmNetwork(id: SupportedCaipChainId): void {
+    if (id === this.state.selectedMultichainNetworkChainId) {
+      if (!this.state.isEvmSelected) {
+        // Same non-EVM network is already selected, no need to update
+        return;
+      }
+
+      // Indicate that the non-EVM network is selected
+      this.update((state) => {
+        state.isEvmSelected = false;
+      });
+
+      // Notify listeners that setActiveNetwork was called
+      this.messagingSystem.publish(
+        'MultichainNetworkController:networkDidChange',
+        id,
+      );
+    }
+
+    // Notify listeners that setActiveNetwork was called
+    this.messagingSystem.publish(
+      'MultichainNetworkController:networkDidChange',
+      id,
+    );
+
+    this.update((state) => {
+      state.selectedMultichainNetworkChainId = id;
+      state.isEvmSelected = false;
+    });
+  }
+
+  /**
+   * Sets the active network.
+   *
+   * @param id - The non-EVM Caip chain ID or EVM client ID of the network to set active.
+   * @returns - A promise that resolves when the network is set active.
+   */
+  async setActiveNetwork(
+    id: SupportedCaipChainId | NetworkClientId,
+  ): Promise<void> {
+    if (isCaipChainId(id)) {
+      const isSupportedCaipChainId = checkIfSupportedCaipChainId(id);
+      if (!isSupportedCaipChainId) {
+        throw new Error(`Unsupported Caip chain ID: ${String(id)}`);
+      }
+      return this.#setActiveNonEvmNetwork(id);
+    }
+
+    return await this.#setActiveEvmNetwork(id);
+  }
+
+  /**
+   * Handles switching between EVM and non-EVM networks when an account is changed
+   *
+   * @param account - The account that was changed
+   */
+  readonly #handleSelectedAccountChange = (account: InternalAccount) => {
+    const { type: accountType, address: accountAddress } = account;
+    const isEvmAccount = isEvmAccountType(accountType);
+
+    // Handle switching to EVM network
+    if (isEvmAccount) {
+      if (this.state.isEvmSelected) {
+        // No need to update if already on evm network
+        return;
+      }
+
+      // Make EVM network active
+      this.update((state) => {
+        state.isEvmSelected = true;
+      });
+      return;
+    }
+
+    // Handle switching to non-EVM network
+    const nonEvmChainId = getChainIdForNonEvmAddress(accountAddress);
+    const isSameNonEvmNetwork =
+      nonEvmChainId === this.state.selectedMultichainNetworkChainId;
+
+    if (isSameNonEvmNetwork) {
+      // No need to update if already on the same non-EVM network
+      this.update((state) => {
+        state.isEvmSelected = false;
+      });
+      return;
+    }
+
+    this.update((state) => {
+      state.selectedMultichainNetworkChainId = nonEvmChainId;
+      state.isEvmSelected = false;
+    });
```

Comment:
Same here, we should add a comment to explain why we are not firing any event in this case.

Reply:
Added